Parse/src/test/java/com/parse/ParseHttpClientTest.java

```diff
@@ -141,4 +182,120 @@ private void doSingleParseHttpClientExecuteWithResponse(int responseCode, String
     // Shutdown mock server
     server.shutdown();
   }
+
+  private void doSingleParseHttpClientExecuteWithGzipResponse(
+      int responseCode, String responseStatus, final String responseContent, ParseHttpClient client)
+      throws Exception {
+    MockWebServer server = new MockWebServer();
+
+    // Make mock response
+    Buffer buffer = new Buffer();
+    ByteArrayOutputStream byteOut = new ByteArrayOutputStream();
+    GZIPOutputStream gzipOut = new GZIPOutputStream(byteOut);
+    gzipOut.write(responseContent.getBytes());
+    gzipOut.close();
+    buffer.write(byteOut.toByteArray());
+    MockResponse mockResponse = new MockResponse()
+        .setStatus("HTTP/1.1 " + responseCode + " " + responseStatus)
+        .setBody(buffer)
+        .setHeader("Content-Encoding", "gzip");
+
+    // Start mock server
+    server.enqueue(mockResponse);
+    server.start();
+
+    // We do not need to add Accept-Encoding header manually, httpClient library should do that.
+    String requestUrl = server.getUrl("/").toString();
+    ParseHttpRequest parseRequest = new ParseHttpRequest.Builder()
+        .setUrl(requestUrl)
+        .setMethod(ParseRequest.Method.GET)
+        .build();
+
+    // Add decompress interceptor
+    client.addInternalInterceptor(new ParseDecompressInterceptor());
+    // Add plain interceptor to disable decompress response stream
+    client.addExternalInterceptor(new ParseNetworkInterceptor() {
+      @Override
+      public ParseHttpResponse intercept(Chain chain) throws IOException {
+        return chain.proceed(chain.getRequest());
+      }
+    });
```

Comment:
It seems like we're testing the `ParseDecompressInterceptor` here and not `ParseHttpClient`. Could we move these new tests into `ParseDecompressInterceptorTest`?

Also, could we verify that `ParseHttpClient` still works with GZIP responses without the help of `ParseDecompressInterceptor`?